src/ale_interface.cpp

```diff
@@ -251,6 +251,36 @@ const ALEScreen& ALEInterface::getScreen() {
   return environment->getScreen();
 }
 
+//This method should receive an empty vector to fill it with
+//the grayscale colours
+void ALEInterface::getScreenGrayscale(std::vector<unsigned char>& grayscale_output_buffer){
+  size_t w = environment->getScreen().width();
+  size_t h = environment->getScreen().height();
+  size_t screen_size = w*h;
+  
+  assert(grayscale_output_buffer.size() == screen_size);
```

Comment:
It's better to not assume that the vector is initialized, and instead allocate the memory using resize(). You can leave the assert() as a sanity check.